sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -1603,6 +1603,39 @@ class CSVSuite extends QueryTest with SharedSQLContext with SQLTestUtils with Te
       .exists(msg => msg.getRenderedMessage.contains("CSV header does not conform to the schema")))
   }
 
+  test("SPARK-25134: check header on parsing of dataset with projection and column pruning") {
```

Comment:
Also need a test case for checking enforceSchema works well when column pruning is on.

Reply:
it seems enforceSchema always kind of "works" because it simply means it ignores the headers.
what do we expect to verify in the test?